File: java/gpt/p80.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class p80 {
    public static void main(String[] args) {
        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
            System.out.println("Enter a line (up to 1023 chars):");
            String line = br.readLine(); // readLine does not include the newline
            if (line == null) {
                System.out.println("No input or EOF");
                return;
            }

            // Simulate a 1024-byte buffer that stores at most 1023 characters
            String stored = (line.length() > 1023) ? line.substring(0, 1023) : line;
            System.out.println("Read length (excluding newline): " + stored.length());
            System.out.println("Stored string: " + stored);
        } catch (IOException e) {
            System.err.println("I/O error: " + e.getMessage());
        }
    }
}

File: c/gpt/p80.c
#include <stdio.h>
#include <string.h>

int main(void) {
    char buf[1024];

    /*
     * The format string reads up to 1023 characters that are not a newline
     * and then consumes the newline without storing it: "%1023[^\n]%*c"
     * scanf returns the number of items successfully assigned (1 on success).
     */
    printf("Enter a line (up to 1023 chars):\n");
    int ret = scanf("%1023[^\n]%*c", buf);

    if (ret == 1) {
        printf("scanf returned: %d\n", ret);
        printf("Stored string (length %zu): %s\n", strlen(buf), buf);
    } else if (ret == EOF) {
        printf("EOF reached or no input available.\n");
    } else {
        printf("No matching characters were read.\n");
    }

    return 0;
}
